.github/workflows/tag-publish.yaml

```diff
@@ -1,10 +1,170 @@
+          role-to-assume: ${{ vars.HOUSTON_UPLOAD_AWS_ROLE }}
+          role-session-name: "houston-upload-${{ github.run_number }}"
+          role-duration-seconds: 900
+      - name: Upload artifacts to Houston
+        # This step is only supported in production as there is no staging version of Houston
+        if: ${{ env.ENVIRONMENT_NAME == 'publish-prod' }}
+        env:
+          HOUSTON_BUCKET: ${{ vars.HOUSTON_BUCKET }}
+        run: |
+          aws s3 sync --acl public-read "$LOCAL_ARTIFACTS_PATH" s3://$HOUSTON_BUCKET/teleport-plugins/${ARTIFACT_TAG##*-v}/ \
+            --include "*" \
+            --exclude "*.tgz*"  # Exclude helm chart artifacts
+      # Image promotion
+      - name: Assume AWS role for publishing the container images
+        uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+        with:
+          role-skip-session-tagging: true
+          aws-region: us-west-2
+          role-to-assume: ${{ vars.CONTAINER_IMAGE_PUBLISHING_SYNC_AWS_ROLE }}
+          role-session-name: "container-image-publishing-sync-${{ github.run_number }}"
+          role-duration-seconds: 900
+      - name: Authenticate with ECR
+        env:
+          CONTAINER_IMAGE_PRIVATE_REGISTRY: ${{ vars.CONTAINER_IMAGE_PRIVATE_REGISTRY }}
+        run: |
+          aws ecr get-login-password | docker login -u="AWS" --password-stdin "$CONTAINER_IMAGE_PRIVATE_REGISTRY"
+          aws ecr-public get-login-password --region us-east-1 | docker login -u="AWS" --password-stdin public.ecr.aws
+      - name: Publish access and event-handler images
+        env:
+          CONTAINER_IMAGE_PRIVATE_REGISTRY: ${{ vars.CONTAINER_IMAGE_PRIVATE_REGISTRY }}
+          CONTAINER_IMAGE_PUBLIC_REGISTRY: ${{ vars.CONTAINER_IMAGE_PUBLIC_REGISTRY}}
+          GITREF: ${{ env.ARTIFACT_TAG }}
+        run: |
+          export VERSION=${ARTIFACT_TAG##v}
+          make -j"$(nproc)" DOCKER_PRIVATE_REGISTRY="$CONTAINER_IMAGE_PRIVATE_REGISTRY" \
+            DOCKER_ECR_PUBLIC_REGISTRY="$CONTAINER_IMAGE_PUBLIC_REGISTRY" docker-promote
+      # Helm promotion
+      - name: Assume AWS role for syncing the Helm artifacts
+        uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+        with:
+          role-skip-session-tagging: true
+          aws-region: us-west-2
+          role-to-assume: ${{ vars.HELM_PUBLISHING_SYNC_AWS_ROLE }}
+          role-session-name: "helm-publishing-sync-${{ github.run_number }}"
+          role-duration-seconds: 900
+      - name: Download the Helm repo from S3
+        env:
+          HELM_REPO_BUCKET: ${{ vars.HELM_REPO_BUCKET }}
+        run: aws s3 cp "s3://$HELM_REPO_BUCKET/" "$LOCAL_HELM_REPO_PATH" --recursive
+      - name: Copy the Helm charts to the repo and regenerate the index
+        working-directory: "${{ env.LOCAL_HELM_REPO_PATH }}"
+        run: |
+          cp $(find "$LOCAL_ARTIFACTS_PATH" -name "teleport-plugin-*.tgz" -type f) .
```

Comment:
I think it is safer to not nest find on the command line this way but instead call cp from find:

    find "$LOCAL_ARTIFACT_PATH" -name "teleport-plugin-*.tgz" -type f -exec cp {} "$(pwd)" \;

Reply:
Agreed, updated